### PR TITLE
[14.0][ADD] delivery_price_product_domain: add new module

### PR DESCRIPTION
**Pull request by solo4games**
Apply domain to product in shipping charges rules

### CODE REVIEW

File: delivery_price_product_domain/models/delivery_carrier.py [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+        for line in order.order_line:
+            qty = line.product_uom._compute_quantity(
+                line.product_uom_qty, line.product_id.uom_id
+            )
+            if line.state == "cancel" or line.product_id not in apply_product:
+                total -= (line.price_unit or 0.0) * qty
+                continue
+            if line.is_delivery:
+                total_delivery += line.price_total
+            if not line.product_id or line.is_delivery:
```
Comment (geomer198) [inline delivery_price_product_domain/models/delivery_carrier.py]: I think better join a conditions in a one.
```
if (not line.product_id or line.is_delivery) and (line.product_id.type == "service"):
    continue
```

File: delivery_price_product_domain/models/delivery_carrier.py [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+                continue
+            if line.product_id.type == "service":
+                continue
+            weight += (line.product_id.weight or 0.0) * qty
+            volume += (line.product_id.volume or 0.0) * qty
+            quantity += qty
+        total += (order.amount_total or 0.0) - total_delivery
+
+        total = self._compute_currency(order, total, "pricelist_to_company")
+        price_dict = self._get_price_dict(total, weight, volume, quantity)
```
Comment (geomer198) [inline delivery_price_product_domain/models/delivery_carrier.py]: Just do like this:
`return self._get_price_dict(total, weight, volume, quantity)`

File: delivery_price_product_domain/models/delivery_carrier.py [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+        return price_dict
+
+    def _get_price_from_picking(self, total, weight, volume, quantity):
+        price_dict = self._get_price_dict(total, weight, volume, quantity)
```
Comment (geomer198) [inline delivery_price_product_domain/models/delivery_carrier.py]: A `price_dict` don't use is next condition will true.
I think better move this line after condition.

**Review by geomer198 — COMMENTED** [review]
Please add a Copyright string in all python files.

For `_get_price_from_picking` and `_get_price_available`  functions please add comments to code

File: delivery_price_product_domain/models/__init__.py [hunk 4]
```diff
@@ -1,2 +1,4 @@
+# Copyright 2023 Cetmix OÜ - Andrey Solodovnikov
```
Comment (rousseldenis) [inline delivery_price_product_domain/models/__init__.py]: @solo4games Copyrights are not necessary in init files

**Review by geomer198 — APPROVED** [review]

**Comment by GSLabIt** [comment]
can you please rebase?

**Review by francesco-ooops — APPROVED** [review]
Functional ok!

We decided to include delivery_price_rule_untaxed among dependencies as it's a useful extra option when managing delivery methods rule, it does not affect standard flow and also to avoid the n-th bridge module development and mainteinance

@rousseldenis what do you think?

**Comment by francesco-ooops** [comment]
@OCA/logistics-maintainers can this be merged? 

tests are failing due to https://github.com/OCA/delivery-carrier/issues/611 so the issue should be temporary

**Comment by francesco-ooops** [comment]
@jbaudoux can I ask you for a review here? thanks!

File: delivery_price_product_domain/models/delivery_carrier.py [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+
+class DeliveryPriceRule(models.Model):
+    _inherit = "delivery.carrier"
+
+    order_id = fields.Many2one("sale.order", string="Sale Order")
+
+    def recompute_price_available(self, apply_product, price_dict, untaxed_in_dict):
+        """This method recompute total parameters only with apply products"""
+        self.ensure_one()
+        self = self.sudo()
```
Comment (rousseldenis) [inline delivery_price_product_domain/models/delivery_carrier.py]: You can't shadow existing variable.

```suggestion
        self_sudo = self.sudo()
```

And the subsequent ones

**Review by rousseldenis — COMMENTED** [review]
LGTM. Technical remark

File: delivery_price_product_domain/models/delivery_carrier.py (hunk 5 above)
Comment (francesco-ooops) [reply delivery_price_product_domain/models/delivery_carrier.py]: @solo4games squash it please

Comment (solo4games) [reply delivery_price_product_domain/models/delivery_carrier.py]: Done

Comment (francesco-ooops) [reply delivery_price_product_domain/models/delivery_carrier.py]: @rousseldenis go? :)

**Pull request merged**